velocidi-sdk/src/test/kotlin/com/velocidi/TrackingEventsTest.kt

```diff
@@ -1,412 +1,310 @@
-    }
-
-    @Test
-    fun customTrackingEventSerialization() {
-        val event = mapOf(
-            "test[a]" to "1",
-            "type" to "custom",
-            "siteId" to "0",
-            "clientId" to "id1"
-        )
-
-        val eventObj = CustomTrackingEvent(
-            "custom",
-            "0",
-            "id1",
-            JSONObject(""" {"test": {"a": 1}} """)
-        )
-
-        assertThat(event).isEqualTo(eventObj.toQueryParams())
-    }
-
-    @Test
-    fun customTrackingEventAppendPropriety() {
-        val event = mapOf(
-            "type" to "custom",
-            "siteId" to "0",
-            "clientId" to "id1",
-            "foo" to "bar"
-        )
-
-        val eventObj = CustomTrackingEvent(
-            "custom",
-            "0",
-            "id1"
-        )
-        eventObj.appendProperty("foo", "bar")
-
-        assertThat(event).isEqualTo(eventObj.toQueryParams())
-    }
+    private val defaultOrderObj = JSONObject(
+        """
+      {
+        "id": "or123",
+        "currency": "EUR",
+        "total": 102.5,
+        "subtotal": 85,
+        "tax": 11.5,
+        "shipping": 6,
+        "discount": {
+          "value": 5,
+          "currency": "EUR"
+        },
+        "refund": 0,
+        "paymentMethod": "Visa",
+        "shippingMethod": "UPS",
+        "shippingCountry": "France",
+        "promotions": [
+          "WINTERSALE"
+        ]
+      }
+        """.trimIndent()
+    )
 
     @Test
     fun pageViewEventSerialization() {
         val event = mapOf(
+            "clientId" to "velocidi",
+            "siteId" to "velocidi.com",
             "type" to "pageView",
-            "siteId" to "0",
-            "clientId" to "0",
-            "location" to "mylocation",
-            "title" to "My page",
+            "customFields[debug]" to "true",
+            "customFields[role]" to "superuser",
+            "title" to "Welcome to your Shop",
             "pageType" to "homepage",
-            "category" to "shopping"
+            "category" to "Shopping"
         )
 
-        val eventObj = PageView(
-            siteId = "0",
-            clientId = "0"
+        val eventObj = JSONObject(
+            """
+            {
+              "clientId": "velocidi",
+              "siteId": "velocidi.com",
+              "type": "pageView",
+              "customFields": {
+                "debug": "true",
+                "role": "superuser"
+              },
+              "title": "Welcome to your Shop",
+              "pageType": "homepage",
+              "category": "Shopping"
+            }
+            """.trimIndent()
         )
-        eventObj.location = "mylocation"
-        eventObj.title = "My page"
-        eventObj.pageType = "homepage"
-        eventObj.category = "shopping"
 
-        assertThat(event).isEqualTo(eventObj.toQueryParams())
+        assertThat(event).containsAllEntriesOf(eventObj.toQueryParams())
```

Comment:
What does this mean, that the first map may contain more entries than the second? If so, shouldn't we be using a more strict matcher throughout the file?

Reply:
Nice catch. The first map could, potentially, contain more elements than the second. AssertJ (assert library used) doesn't provide a matcher that does what we want so I changed this spec to use a custom matcher everywhere. Check 51ec0d7

Reply:
I find it strange we don't have that matcher in the core library. We have `containsExactlyEntriesOf` which considers order though... I guess this will do.